packages/openhim/test/index.js

```diff
@@ -45,33 +69,326 @@ describe('execute', () => {
   });
 });
 
-describe.skip('event', () => {
-  let mockRequest;
+describe('encounter', () => {
+  it('creates an encounter with given data', async () => {
+    testServer
+      .intercept({
+        path: '/chw/encounter',
+        method: 'POST',
+      })
+      .reply(201, 'User Authenticated Successfully', { ...jsonHeaders });
+
+    const { data } = await encounter({
+      resourceType: 'Patient',
+      id: 'example',
+      identifier: [
+        {
+          fhir_comments: [
+            '   MRN assigned by ACME healthcare on 6-May 2001   ',
+          ],
+          use: 'usual',
+          type: {
+            coding: [
+              {
+                system: 'http://hl7.org/fhir/v2/0203',
+                code: 'MR',
+              },
+            ],
+          },
+          system: 'urn:oid:1.2.36.146.595.217.0.1',
+          value: '12345',
+          period: {
+            start: '2001-05-06',
+          },
+          assigner: {
+            display: 'Acme Healthcare',
+          },
+        },
+      ],
+      active: true,
 
-  before(() => {
-    mockRequest = superagentMock(request, ClientFixtures);
-  });
+      deceasedBoolean: false,
 
-  it('posts to API and returns state', () => {
-    let state = {
-      configuration: {
-        username: 'hello',
-        password: 'there',
-        apiUrl: 'https://openhim.org/demo',
+      managingOrganization: {
+        reference: 'Organization/1',
       },
-    };
+    })(state);
+    expect(data).to.eql('User Authenticated Successfully');
+  });
+});
+
+describe('getTransactions', () => {
+  it('should GET all transctions', async () => {
+    testServer
+      .intercept({
+        path: '/transactions',
+        method: 'GET',
+      })
+      .reply(200, [{ _id: '001', clientID: '1234' }], { ...jsonHeaders });
 
-    return execute(event(fixtures.event.requestBody))(state).then(state => {
-      let lastReference = state.references[0];
+    const { data } = await getTransactions()(state);
+    expect(data[0].clientID).to.eql('1234');
+  });
 
-      // Check that the eventData made it's way to the request as a string.
-      expect(lastReference.params).to.eql(
-        JSON.stringify(fixtures.event.requestBody)
+  it('should GET a single transaction', async () => {
+    testServer
+      .intercept({
+        path: '/transactions/001',
+        method: 'GET',
+      })
+      .reply(
+        200,
+        {
+          _id: '001',
+          clientID: '1234',
+        },
+        { ...jsonHeaders }
       );
-    });
+
+    const { data } = await getTransactions({ transactionId: '001' })(state);
+    expect(data.clientID).to.eql('1234');
+    expect(data._id).to.eql('001');
+  });
+
+  it('should GET transactions with filters', async () => {
```

Comment:
Can we have a test that undocumented query options get appended to the url? Like `getTransactions({ foo: 'bar' })` should generate the correct URL